shared/checks/oval/installed_app_is_ocp4.xml

```diff
@@ -10,19 +10,6 @@
     </metadata>
     <criteria>
       <extend_definition comment="RHEL8 OS installed" definition_ref="installed_OS_is_rhel8" />
-      <criterion comment="OpenShift Node is installed" test_ref="test_ocp4_node" />
```

Comment:
You still need to check if OCP4 is installed. Checking just the OS install is not enough.

Reply:
@redhatrises we don't support running RHCOS standalone. So OC4 has to be there.

Reply:
@JAORMX agreed. That is my original statement in this part of the review.

Reply:
If you tell me how exactly this node can be detected, I'd happily add this check.

Reply:
I mean *beside* the obvious way of copying the checks from ocp3 and replacing the number, because RPM-based checks would not work.

Reply:
Off the top of my head the only way I'm aware of that would be representative to if the node was _actively_ in a cluster would be to use `oc` itself. More or less `oc status` and check the return code. However, RHCOS isn't used in any other context other than OCP and that's by design which is why OCP 4 components, such as the cri-o runtime, mcd, and kubelet are baked in.

Reply:
Thanks for chiming in folks! If checking `oc status` is acceptable then let's use that to check the node. If that's not an acceptable solution, I'm fine with the PR merging as-is.

Reply:
If we don't support RHCOS in other context than OCP4 then why is the additional check required? Future-proofing?

Reply:
The content needs to check the following and oc status doesn't work without an `oc` probe last I checked:

1. RHEL version
2. CoreOS version
3. OCP version installed on CoreOS
4. OCP version installed on RHEL node

Reply:
Correct. For `oc` to work the node must be in the cluster. I believe the following would get you the information on both:
- Version: `VERSION` in /etc/os-release
- OCP version: Check `openshift-clients` package
```
rpm -q --qf "%{VERSION}" openshift-clients
```

Reply:
Why can't we use the OCP version from the VERSION line in /etc/os-release again? We're already parsing that file..
VERSION="42.80.20191022.0 => OCP version=4.2, RHCOS version=8.0

Reply:
Because that actually doesn't guarantee that the openshift bits are installed... just coreos, and it also won't work on a RHEL node.

Reply:
Correct. The bits will be there on RHCOS as it's a component of Openshift, but RHEL may not have them since RHEL is general purpose.

Reply:
Still doesn't matter if the bits are already installed on CoreOS because CoreOS is going to have a CPE (which is currently missing) that isn't the OpenShift CPE. CPE is part of this which is why just checking `/etc/os-release` isn't correct for checking the OCP parts of the check. In fact when CoreOS gets a CPE, we are going to have to split out the coreos installed check anyway.

Reply:
In fact while we are having this discussion, there is going to have to be a way to validate with a gpg check not only the software installed in CoreOS, but also a container that is installed on coreos.

Reply:
> In fact while we are having this discussion, there is going to have to be a way to validate with a gpg check not only the software installed in CoreOS, but also a container that is installed on coreos.

Reply:
Still have to validate regardless of source against a key to meet compliance requirements.

Reply:
I say we merge this for now so we can start using this to build content already. And we start iterating on this as we go. I'm fine with merging this as it is.

Reply:
I understand where you're coming from and especially the points about RHEL nodes are valid. But we need to start working on the content and even just RHCOS-based is good enough.

On the RHCOS nodes: just knowing that "a node is RHCOS" is good enough at least for now because 1) RHCOS only ever exists as an OCP node 2) The content is supposed to be used from an operator anyway which can only run in OCP.

Again, I agree with your points. But if possible, I would prefer to merge the stub of the content ASAP and then iterate on it to provide a technically more sound solution.